internal/support/scheduler/init.go

```diff
@@ -0,0 +1,265 @@
+				}
+				// Setup Logging
+				logTarget := setLoggingTarget()
+				LoggingClient = logger.NewClient(internal.SupportSchedulerServiceKey, Configuration.Logging.EnableRemote, logTarget)
+			}
+		}
+
+		if Configuration != nil {
+			break
+		}
+		time.Sleep(time.Second * time.Duration(1))
+	}
+	close(ch)
+	wait.Done()
+
+	return
+}
+
+func Init(useConsul bool) bool {
+	if Configuration == nil {
+		return false
+	}
+
+	if useConsul {
+		chConfig = make(chan interface{})
+		go listenForConfigChanges()
+	}
+	return true
+}
+
+func Destruct() {
+	if chConfig != nil {
+		close(chConfig)
+	}
+}
+
+func initializeConfiguration(useConsul bool, useProfile string) (*ConfigurationStruct, error) {
+	//We currently have to load configuration from filesystem first in order to obtain ConsulHost/Port
+	conf := &ConfigurationStruct{}
+	err := config.LoadFromFileV2(useProfile, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	if useConsul {
+		conf, err = connectToConsul(conf)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return conf, nil
+}
+
+func connectToConsul(conf *ConfigurationStruct) (*ConfigurationStruct, error) {
+	//Obtain ConsulConfig
+	cfg := consulclient.NewConsulConfig(conf.Registry, conf.Service, internal.SupportSchedulerServiceKey)
+
+	// Register the service in Consul
+	err := consulclient.ConsulInit(cfg)
+
+	if err != nil {
+		return conf, fmt.Errorf("connection to Consul could not be made: %v", err.Error())
+	}
+	// Update configuration data from Consul
+	updateCh := make(chan interface{})
+	errCh := make(chan error)
+	dec := consulclient.NewConsulDecoder(conf.Registry)
+	dec.Target = &ConfigurationStruct{}
+	dec.Prefix = internal.ConfigV2Stem + internal.SupportSchedulerServiceKey
+	dec.ErrCh = errCh
+	dec.UpdateCh = updateCh
+
+	defer dec.Close()
+	defer close(updateCh)
+	defer close(errCh)
+	go dec.Run()
+
+	select {
+	case <-time.After(2 * time.Second):
+		err = errors.New("timeout loading config from registry")
+	case ex := <-errCh:
+		err = errors.New(ex.Error())
+	case raw := <-updateCh:
+		actual, ok := raw.(*ConfigurationStruct)
+		if !ok {
+			return conf, errors.New("type check failed")
+		}
+		conf = actual
+	}
+
+	return conf, err
+}
+
+func listenForConfigChanges() {
+	errCh := make(chan error)
+	dec := consulclient.NewConsulDecoder(Configuration.Registry)
+	dec.Target = &ConfigurationStruct{}
+	dec.Prefix = internal.ConfigV2Stem + internal.SupportSchedulerServiceKey
+	dec.ErrCh = errCh
+	dec.UpdateCh = chConfig
+
+	defer dec.Close()
+	defer close(errCh)
+
+	go dec.Run()
+	for {
+		select {
+		case ex := <-errCh:
+			LoggingClient.Error(ex.Error())
+		case raw, ok := <-chConfig:
+			if ok {
+				actual, ok := raw.(*ConfigurationStruct)
+				if !ok {
+					LoggingClient.Error("listenForConfigChanges() type check failed")
+				}
+				Configuration = actual //Mutex needed?
+			} else {
+				return
+			}
+		}
+	}
+}
+
+func initializeClients(useConsul bool) {
+	// Create metadata clients
```

Comment:
This is currently an unused function. Whether it remains that way hinges on the question regarding which service client this application should be using.

Reply:
removed this function as were not currently using this function.